### PR TITLE
Migrate view items to native view bindings

### LINKED ISSUE #1829: use seperate xml designs for Items (+use viewbinding)
UserItem.java uses 2 different xml designs which adds unnecessary complexity. It's then used in ConversationInforController and ContactsController.

There should be seperate Items for each use case. This is also required to switch from Butterknife to viewbinding.

assigned @AndyScherzinger as we discussed it while talking about https://github.com/nextcloud/talk-android/pull/1668

### PR DESCRIPTION
**Pull request by AndyScherzinger**
* Resolves #1829
* Migrates any view item to a native view binding

Signed-off-by: Andy Scherzinger <info@andy-scherzinger.de>

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
❗ rebased to cut down on the drone commits

**Review by mahibi — APPROVED** [review]

File: app/src/main/java/com/nextcloud/talk/adapters/items/UserItem.java [hunk 1]
```diff
@@ -62,20 +52,18 @@
...
-    private Participant participant;
-    private UserEntity userEntity;
+    private final Participant participant;
+    private final UserEntity userEntity;
     private GenericTextHeaderItem header;
     public boolean isOnline = true;
 
-    public UserItem(Context activityContext,
-                    Participant participant,
+    public UserItem(Participant participant,
```
Comment (mahibi) [inline app/src/main/java/com/nextcloud/talk/adapters/items/UserItem.java:64]: maybe we should rename UserItem to ContactItem (according to R.layout.rv_item_contact and ContactsController)

File: app/src/main/java/com/nextcloud/talk/adapters/items/AppItem.java [hunk 2]
```diff
@@ -25,26 +25,25 @@
...
+
+import androidx.annotation.Nullable;
 import eu.davidea.flexibleadapter.FlexibleAdapter;
 import eu.davidea.flexibleadapter.items.AbstractFlexibleItem;
 import eu.davidea.flexibleadapter.items.IFlexible;
 import eu.davidea.viewholders.FlexibleViewHolder;
 
-import java.util.List;
-
 public class AppItem extends AbstractFlexibleItem<AppItem.AppItemViewHolder> {
```
Comment (mahibi) [inline app/src/main/java/com/nextcloud/talk/adapters/items/AppItem.java:41]: AppItem is not used anywehere...?

File: app/src/main/java/com/nextcloud/talk/adapters/items/MenuItem.java [hunk 3]
```diff
@@ -22,36 +22,38 @@
...
+
+import java.util.List;
+
 import eu.davidea.flexibleadapter.FlexibleAdapter;
 import eu.davidea.flexibleadapter.items.AbstractFlexibleItem;
 import eu.davidea.viewholders.FlexibleViewHolder;
 
-import java.util.List;
-
 public class MenuItem extends AbstractFlexibleItem<MenuItem.MenuItemViewHolder> {
```
Comment (mahibi) [inline app/src/main/java/com/nextcloud/talk/adapters/items/MenuItem.java:42]: not used anywhere

File: app/src/main/java/com/nextcloud/talk/adapters/items/ProgressItem.java [hunk 4]
```diff
@@ -71,8 +70,8 @@ public int getLayoutRes() {
     @Override
```
Comment (mahibi) [inline app/src/main/java/com/nextcloud/talk/adapters/items/ProgressItem.java:70]: ProgressItem is also not used anywhere :D

File: app/src/main/java/com/nextcloud/talk/adapters/items/AppItem.java (hunk 2 above)
Comment (mahibi) [reply app/src/main/java/com/nextcloud/talk/adapters/items/AppItem.java:41]: this one should have been deleted with
https://github.com/nextcloud/talk-android/pull/1285/files
as i simplified the "share to" feature..

**Comment by AndyScherzinger** [comment]
Thanks for the review, I'll take care of the findings 👍

**Comment by AndyScherzinger** [comment]
@mahibi fixed all your comments, thanks gain. So let's see what CI says and if the code checks lead to an even better result 🤞

**Comment by AndyScherzinger** [comment]
did some more cleanup, since the removal of unused classes led to new removals of the corresponding layouts and their string resources 🎉

**Pull request merged**